src/hazelcore/semantics/Action.re

```diff
@@ -3422,6 +3579,64 @@ let rec syn_perform_block =
     let (ze, u_gen) = ZExp.new_EmptyHole(u_gen);
     let zblock = ZExp.BlockZE(lines, ze);
     syn_perform_block(~ci, ctx, keyword_action(k), (zblock, Hole, u_gen));
+
+  /* Variable Splitting */
+  | (Construct(SOp(SSpace)), BlockZE(lines, zexp))
+      when is_zexp_split_on_keyword(zexp) =>
+    switch (ZExp.cursor_on_var(zexp)) {
+    | Some((pos, _, _, name)) =>
+      let keyword =
+        String.sub(name, 0, pos)
+        |> Keyword.of_string
+        |> GeneralUtil.Opt.get_exn;
+      let var = String.sub(name, pos, String.length(name) - pos);
+      let (ze, u_gen) = ZExp.new_EmptyHole(u_gen);
+      let action = Construct(SVar(var, OnText(String.length(name) - pos)));
```

Comment:
Let's leave the cursor at `OnText(0)` since that's where the cursor would end up if user actually inserted a text space